odoo/tools/safe_eval.py

```diff
@@ -240,28 +240,24 @@ def assert_valid_codeobj(allowed_codes, code_obj, expr):
             assert_valid_codeobj(allowed_codes, const, 'lambda')
 
 
-def test_expr(expr, allowed_codes, mode="eval", filename=None):
-    """test_expr(expression, allowed_codes[, mode[, filename]]) -> code_object
-
-    Test that the expression contains only the allowed opcodes.
-    If the expression is valid and contains only allowed codes,
-    return the compiled code object.
-    Otherwise raise a ValueError, a Syntax Error or TypeError accordingly.
-
-    :param filename: optional pseudo-filename for the compiled expression,
-                 displayed for example in traceback frames
-    :type filename: string
+def compile_codeobj(expr, /, filename=None, mode='eval'):
```

Comment:
```suggestion
def compile_codeobj(expr, /, filename='<unknown>', mode='eval'):
```

You could put types in the signature too.